### PR TITLE
[Dynamic Instrumentation] DEBUG-2524 Fix EL numeric binary operations

### PR DESCRIPTION
**Pull request by dudikeleti**
## Summary of changes
This PR fix the process of comparison different numeric types

## Test coverage
Updated EL tests & system tests

### CODE REVIEW

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (5815)](https://github.com/DataDog/dd-trace-dotnet/tree/4b3b489cf7e7b50a3cf71303ad5e1feef359b84f)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/8b5987de4fa4db6da24d470fc31439bae6a59b45)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5815) - mean (73ms)  : 64, 83
     .   : milestone, 73,
    master - mean (72ms)  : 62, 82
     .   : milestone, 72,

    section CallTarget+Inlining+NGEN
    This PR (5815) - mean (1,060ms)  : 1038, 1082
     .   : milestone, 1060,
    master - mean (1,057ms)  : 1037, 1078
     .   : milestone, 1057,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5815) - mean (110ms)  : 106, 115
     .   : milestone, 110,
    master - mean (109ms)  : 106, 112
     .   : milestone, 109,

    section CallTarget+Inlining+NGEN
    This PR (5815) - mean (754ms)  : 731, 776
     .   : milestone, 754,
    master - mean (758ms)  : 735, 781
     .   : milestone, 758,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5815) - mean (93ms)  : 90, 96
     .   : milestone, 93,
    master - mean (93ms)  : 89, 97
     .   : milestone, 93,

    section CallTarget+Inlining+NGEN
    This PR (5815) - mean (702ms)  : 679, 725
     .   : milestone, 702,
    master - mean (702ms)  : 680, 724
     .   : milestone, 702,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5815) - mean (194ms)  : 190, 198
     .   : milestone, 194,
    master - mean (192ms)  : 189, 196
     .   : milestone, 192,

    section CallTarget+Inlining+NGEN
    This PR (5815) - mean (1,166ms)  : 1114, 1219
     .   : milestone, 1166,
    master - mean (1,165ms)  : 1122, 1207
     .   : milestone, 1165,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5815) - mean (277ms)  : 272, 281
     .   : milestone, 277,
    master - mean (276ms)  : 271, 281
     .   : milestone, 276,

    section CallTarget+Inlining+NGEN
    This PR (5815) - mean (928ms)  : 871, 985
     .   : milestone, 928,
    master - mean (915ms)  : 881, 949
     .   : milestone, 915,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (5815) - mean (266ms)  : 262, 271
     .   : milestone, 266,
    master - mean (266ms)  : 262, 269
     .   : milestone, 266,

    section CallTarget+Inlining+NGEN
    This PR (5815) - mean (965ms)  : 703, 1228
     .   : milestone, 965,
    master - mean (901ms)  : 880, 922
     .   : milestone, 901,

```

**Comment by andrewlock** [comment]
## Benchmarks Report for tracer :snail:

Benchmarks for #5815 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9): 
* 2 benchmarks are **faster**, with geometric mean 1.135
* 2 benchmarks are **slower**, with geometric mean 1.122
* 1 benchmarks have **more** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.ActivityBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartStopWithChild`|net6.0|7.78μs|41.9ns|233ns|0.0115| 0.00382|0|5.43&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartStopWithChild`|netcoreapp3.1|10.1μs|54.8ns|310ns|0.0195| 0.00975|0|5.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartStopWithChild`|net472|16.1μs|56ns|217ns|1.02| 0.3|0.0972|6.07&nbsp;KB|
|#5815|`StartStopWithChild`|net6.0|7.78μs|44.9ns|359ns|0.0155| 0.00776|0|5.43&nbsp;KB|
|#5815|`StartStopWithChild`|netcoreapp3.1|9.85μs|56ns|396ns|0.0146| 0.00488|0|5.62&nbsp;KB|
|#5815|`StartStopWithChild`|net472|15.8μs|49.7ns|193ns|1.01| 0.298|0.0942|6.07&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`WriteAndFlushEnrichedTraces`|net6.0|473μs|282ns|1.09μs|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|645μs|488ns|1.89μs|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`WriteAndFlushEnrichedTraces`|net472|855μs|875ns|3.28μs|0.425| 0|0|3.3&nbsp;KB|
|#5815|`WriteAndFlushEnrichedTraces`|net6.0|477μs|256ns|925ns|0| 0|0|2.7&nbsp;KB|
|#5815|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|646μs|546ns|2.11μs|0| 0|0|2.7&nbsp;KB|
|#5815|`WriteAndFlushEnrichedTraces`|net472|845μs|440ns|1.71μs|0.419| 0|0|3.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendRequest`|net6.0|201μs|1.17μs|10.7μs|0.195| 0|0|18.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendRequest`|netcoreapp3.1|224μs|1.32μs|12.2μs|0.235| 0|0|20.61&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendRequest`|net472|0.00348ns|0.00114ns|0.00426ns|0| 0|0|0&nbsp;b|
|#5815|`SendRequest`|net6.0|191μs|1.02μs|5.92μs|0.196| 0|0|18.45&nbsp;KB|
|#5815|`SendRequest`|netcoreapp3.1|221μs|1.28μs|10.4μs|0.212| 0|0|20.61&nbsp;KB|
|#5815|`SendRequest`|net472|0.000395ns|0.000316ns|0.00118ns|0| 0|0|0&nbsp;b|
</details>


<details><summary><strong>Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark</strong> - Same speed :heavy_check_mark: More allocations :warning:



#### More allocations :warning: in #5815

|  Benchmark         | Base Allocated | Diff Allocated | Change   | Change % |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark.WriteAndFlushEnrichedTraces&#8209;net6.0 | 41.56 KB | 41.77 KB | 210 B | 0.51%

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`WriteAndFlushEnrichedTraces`|net6.0|562μs|2.64μs|10.9μs|0.558| 0|0|41.56&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|689μs|3.46μs|16.6μs|0.34| 0|0|41.74&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`WriteAndFlushEnrichedTraces`|net472|849μs|3.26μs|12.6μs|8.39| 2.52|0.419|53.27&nbsp;KB|
|#5815|`WriteAndFlushEnrichedTraces`|net6.0|571μs|3μs|19.5μs|0.579| 0|0|41.77&nbsp;KB|
|#5815|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|679μs|3.28μs|13.9μs|0.331| 0|0|41.76&nbsp;KB|
|#5815|`WriteAndFlushEnrichedTraces`|net472|886μs|3.8μs|14.7μs|8.42| 2.66|0.443|53.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Faster :tada: Same allocations :heavy_check_mark:



#### Faster :tada: in #5815

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.DbCommandBenchmark.ExecuteNonQuery&#8209;net6.0 | 1.128 | 1,371.06 | 1,215.79 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`ExecuteNonQuery`|net6.0|1.37μs|1.11ns|4.32ns|0.0144| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`ExecuteNonQuery`|netcoreapp3.1|1.66μs|1.14ns|4.26ns|0.0132| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`ExecuteNonQuery`|net472|2.04μs|2.59ns|10ns|0.156| 0|0|987&nbsp;B|
|#5815|`ExecuteNonQuery`|net6.0|1.22μs|1.18ns|4.4ns|0.014| 0|0|1.02&nbsp;KB|
|#5815|`ExecuteNonQuery`|netcoreapp3.1|1.71μs|1.09ns|4.07ns|0.0136| 0|0|1.02&nbsp;KB|
|#5815|`ExecuteNonQuery`|net472|2.04μs|1.51ns|5.83ns|0.157| 0|0|987&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`CallElasticsearch`|net6.0|1.15μs|0.723ns|2.7ns|0.0136| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`CallElasticsearch`|netcoreapp3.1|1.59μs|0.58ns|2.17ns|0.0132| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`CallElasticsearch`|net472|2.51μs|1.36ns|5.26ns|0.158| 0|0|995&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`CallElasticsearchAsync`|net6.0|1.3μs|6.34ns|26.1ns|0.0134| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`CallElasticsearchAsync`|netcoreapp3.1|1.64μs|1.14ns|4.43ns|0.014| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`CallElasticsearchAsync`|net472|2.7μs|2.1ns|7.85ns|0.166| 0.00134|0|1.05&nbsp;KB|
|#5815|`CallElasticsearch`|net6.0|1.17μs|0.84ns|3.14ns|0.0136| 0|0|976&nbsp;B|
|#5815|`CallElasticsearch`|netcoreapp3.1|1.52μs|1.39ns|5.18ns|0.013| 0|0|976&nbsp;B|
|#5815|`CallElasticsearch`|net472|2.57μs|5.06ns|18.2ns|0.158| 0.00129|0|995&nbsp;B|
|#5815|`CallElasticsearchAsync`|net6.0|1.26μs|0.579ns|2.17ns|0.0132| 0|0|952&nbsp;B|
|#5815|`CallElasticsearchAsync`|netcoreapp3.1|1.62μs|1.33ns|4.97ns|0.0138| 0|0|1.02&nbsp;KB|
|#5815|`CallElasticsearchAsync`|net472|2.72μs|2.06ns|8ns|0.166| 0|0|1.05&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`ExecuteAsync`|net6.0|1.19μs|0.492ns|1.91ns|0.0132| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`ExecuteAsync`|netcoreapp3.1|1.61μs|0.689ns|2.58ns|0.0127| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`ExecuteAsync`|net472|1.74μs|0.916ns|3.43ns|0.145| 0|0|915&nbsp;B|
|#5815|`ExecuteAsync`|net6.0|1.2μs|1.03ns|3.86ns|0.0132| 0|0|952&nbsp;B|
|#5815|`ExecuteAsync`|netcoreapp3.1|1.62μs|1.38ns|5.35ns|0.0132| 0|0|952&nbsp;B|
|#5815|`ExecuteAsync`|net472|1.73μs|0.665ns|2.4ns|0.145| 0.000869|0|915&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendAsync`|net6.0|4.29μs|2.3ns|8.3ns|0.0301| 0|0|2.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendAsync`|netcoreapp3.1|5.07μs|2.09ns|8.08ns|0.0362| 0|0|2.76&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendAsync`|net472|7.69μs|2.92ns|11.3ns|0.496| 0|0|3.15&nbsp;KB|
|#5815|`SendAsync`|net6.0|4.16μs|3.5ns|13.1ns|0.0314| 0|0|2.22&nbsp;KB|
|#5815|`SendAsync`|netcoreapp3.1|5.3μs|4.62ns|17.3ns|0.0369| 0|0|2.76&nbsp;KB|
|#5815|`SendAsync`|net472|7.81μs|2.62ns|10.1ns|0.498| 0|0|3.15&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #5815

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.ILoggerBenchmark.EnrichedLog&#8209;netcoreapp3.1 | 1.126 | 2,058.78 | 2,317.23 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|net6.0|1.52μs|0.572ns|2.14ns|0.0227| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|netcoreapp3.1|2.06μs|0.551ns|1.99ns|0.0218| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|net472|2.68μs|4.24ns|16.4ns|0.249| 0|0|1.57&nbsp;KB|
|#5815|`EnrichedLog`|net6.0|1.51μs|0.966ns|3.62ns|0.0233| 0|0|1.64&nbsp;KB|
|#5815|`EnrichedLog`|netcoreapp3.1|2.32μs|1.52ns|5.89ns|0.0219| 0|0|1.64&nbsp;KB|
|#5815|`EnrichedLog`|net472|2.69μs|1.73ns|6.71ns|0.249| 0|0|1.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|net6.0|116μs|154ns|598ns|0| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|netcoreapp3.1|120μs|139ns|540ns|0| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|net472|147μs|97.1ns|376ns|0.657| 0.219|0|4.46&nbsp;KB|
|#5815|`EnrichedLog`|net6.0|116μs|105ns|408ns|0.058| 0|0|4.28&nbsp;KB|
|#5815|`EnrichedLog`|netcoreapp3.1|120μs|232ns|898ns|0.0598| 0|0|4.28&nbsp;KB|
|#5815|`EnrichedLog`|net472|149μs|138ns|533ns|0.673| 0.224|0|4.46&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|net6.0|3.06μs|1.46ns|5.48ns|0.0309| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|netcoreapp3.1|4.21μs|1.39ns|5.37ns|0.0294| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|net472|4.76μs|1.1ns|4.25ns|0.321| 0|0|2.02&nbsp;KB|
|#5815|`EnrichedLog`|net6.0|3.14μs|1.26ns|4.53ns|0.0312| 0|0|2.2&nbsp;KB|
|#5815|`EnrichedLog`|netcoreapp3.1|4.13μs|1.11ns|4.14ns|0.0296| 0|0|2.2&nbsp;KB|
|#5815|`EnrichedLog`|net472|4.76μs|1.18ns|4.57ns|0.319| 0|0|2.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendReceive`|net6.0|1.33μs|1.12ns|4.34ns|0.0158| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendReceive`|netcoreapp3.1|1.71μs|1.16ns|4.51ns|0.0156| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`SendReceive`|net472|2.11μs|0.868ns|3.36ns|0.183| 0.00105|0|1.16&nbsp;KB|
|#5815|`SendReceive`|net6.0|1.37μs|0.607ns|2.35ns|0.0157| 0|0|1.14&nbsp;KB|
|#5815|`SendReceive`|netcoreapp3.1|1.8μs|0.492ns|1.78ns|0.0153| 0|0|1.14&nbsp;KB|
|#5815|`SendReceive`|net472|2.06μs|0.672ns|2.51ns|0.183| 0|0|1.16&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|net6.0|2.73μs|0.984ns|3.81ns|0.0218| 0|0|1.6&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|netcoreapp3.1|3.95μs|1.85ns|6.68ns|0.0222| 0|0|1.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`EnrichedLog`|net472|4.45μs|4.31ns|16.7ns|0.323| 0|0|2.04&nbsp;KB|
|#5815|`EnrichedLog`|net6.0|2.68μs|0.942ns|3.52ns|0.0215| 0|0|1.6&nbsp;KB|
|#5815|`EnrichedLog`|netcoreapp3.1|4.09μs|2.73ns|10.6ns|0.0224| 0|0|1.65&nbsp;KB|
|#5815|`EnrichedLog`|net472|4.4μs|2.28ns|8.54ns|0.322| 0|0|2.04&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #5815

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SpanBenchmark.StartFinishSpan&#8209;net472 | 1.118 | 582.56 | 651.19 | 



#### Faster :tada: in #5815

|  Benchmark         | base/diff | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.SpanBenchmark.StartFinishScope&#8209;netcoreapp3.1 | 1.143 | 759.97 | 664.84 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartFinishSpan`|net6.0|403ns|0.111ns|0.432ns|0.00811| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartFinishSpan`|netcoreapp3.1|588ns|0.482ns|1.87ns|0.00768| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartFinishSpan`|net472|582ns|0.169ns|0.655ns|0.0918| 0|0|578&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartFinishScope`|net6.0|634ns|0.18ns|0.674ns|0.00991| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartFinishScope`|netcoreapp3.1|760ns|0.45ns|1.74ns|0.00945| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`StartFinishScope`|net472|909ns|1.73ns|6.47ns|0.104| 0|0|658&nbsp;B|
|#5815|`StartFinishSpan`|net6.0|400ns|0.104ns|0.401ns|0.00804| 0|0|576&nbsp;B|
|#5815|`StartFinishSpan`|netcoreapp3.1|549ns|0.16ns|0.6ns|0.00773| 0|0|576&nbsp;B|
|#5815|`StartFinishSpan`|net472|651ns|0.533ns|2.07ns|0.0918| 0|0|578&nbsp;B|
|#5815|`StartFinishScope`|net6.0|618ns|0.484ns|1.87ns|0.00988| 0|0|696&nbsp;B|
|#5815|`StartFinishScope`|netcoreapp3.1|665ns|0.783ns|3.03ns|0.00931| 0|0|696&nbsp;B|
|#5815|`StartFinishScope`|net472|823ns|0.165ns|0.617ns|0.104| 0|0|658&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`RunOnMethodBegin`|net6.0|599ns|0.162ns|0.605ns|0.00982| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`RunOnMethodBegin`|netcoreapp3.1|961ns|1.15ns|4.47ns|0.00951| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/3f94ea60126aa3ef652d6620609be329f11ef5e9)|`RunOnMethodBegin`|net472|1.14μs|0.303ns|1.17ns|0.104| 0|0|658&nbsp;B|
|#5815|`RunOnMethodBegin`|net6.0|599ns|0.292ns|1.13ns|0.00963| 0|0|696&nbsp;B|
|#5815|`RunOnMethodBegin`|netcoreapp3.1|901ns|0.399ns|1.55ns|0.00948| 0|0|696&nbsp;B|
|#5815|`RunOnMethodBegin`|net472|1.14μs|0.48ns|1.86ns|0.105| 0|0|658&nbsp;B|
</details>

File: tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.General.cs [hunk 1]
```diff
@@ -19,63 +19,81 @@ internal partial class ProbeExpressionParser<T>
 {
     private const BindingFlags GetMemberFlags = BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.Instance | BindingFlags.FlattenHierarchy;
 
-    private static Expression ConvertToDouble(Expression expr)
+    // https://learn.microsoft.com/en-us/dotnet/standard/base-types/conversion-tables
+    private static Type GetWiderType(Type left, Type right)
```
Comment (GreenMatan) [inline tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.General.cs:23]: Nit: `GetWiderNumericType`

**Review by GreenMatan — APPROVED** [review]
LGTM. We should start evaluating an approach to find all the edge cases using a variation of exploration testing for expression language.

**Review by bouwkast — APPROVED** [review]
👍

**Pull request merged**